### PR TITLE
Detect package version change even when SHA stays the same

### PR DESCRIPTION
**Pull request by premun**
Resolves a problem found in https://github.com/dotnet/arcade-services/issues/3010

When we get 2 builds from the same SHA, the VMR would not update the package version in its metadata as it would assume no changes are needed.

### Release Note Category
- [ ] Feature changes/additions 
- [x] Bug fixes
- [ ] Internal Infrastructure Improvements

### Release Note Description
Detect package version change even when SHA stays the same

### CODE REVIEW

File: src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs [hunk 1]
```diff
@@ -182,10 +172,34 @@ public class VmrUpdater : VmrManagerBase, IVmrUpdater
...
+            currentVersion.Sha,
+            update.RemoteUri,
+            update.TargetRevision);
 
-        _logger.LogInformation("Synchronizing {name} from {current} to {repo} / {revision}{oneByOne}",
-            update.Mapping.Name, currentSha, update.RemoteUri, update.TargetRevision, noSquash ? " one commit at a time" : string.Empty);
+        // Do we need to change anything?
+        if (currentVersion.Sha == update.TargetRevision)
+        {
+            if (currentVersion.PackageVersion != update.TargetVersion && update.TargetVersion != null)
```
Comment (dkurepa) [inline src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs:187]: nit: should the order of these conditions be swapped? Think the compiler will make it so we don't look at the second one if `update.TargetVersion == null` in that case

File: src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs [hunk 2]
```diff
@@ -653,6 +659,43 @@ private void DeleteRepository(string repo)
...
+    /// differs. This can happen when a new build from the same commit is synchronized to the VMR.
+    /// </summary>
+    private async Task UpdateTargetVersionOnly(
+        string targetRevision,
+        string targetVersion,
+        SourceMapping mapping,
+        VmrDependencyVersion currentVersion,
+        CancellationToken cancellationToken)
+    {
+        _logger.LogWarning(
```
Comment (dkurepa) [inline src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs:673]: Should this be a warning? It's not a case we see very often, but it's still something that's a pretty normal thing

File: src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs (hunk 1 above)
Comment (premun) [reply src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs:187]: Yeah but actually the first one will be false more often than the second so this micro-optimization doesn't make sense even in that regard

File: src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs (hunk 2 above)
Comment (premun) [reply src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs:673]: Yeah, I imagine we'd want to learn about this in the build. We were just chasing this down so I think this should be visible

**Review by dkurepa — APPROVED** [review]

**Pull request merged**